src/main/java/uk/gov/hmcts/ccd/security/idam/IdamRepository.java

```diff
@@ -0,0 +1,23 @@
+package uk.gov.hmcts.ccd.security.idam;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.cache.annotation.Cacheable;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.idam.client.IdamClient;
+import uk.gov.hmcts.reform.idam.client.models.UserInfo;
+
+@Component
+public class IdamRepository {
+
+    private final IdamClient idamClient;
+
+    @Autowired
+    public IdamRepository(IdamClient idamClient) {
+        this.idamClient = idamClient;
+    }
+
+    @Cacheable(value = "userInfoCache")
```

Comment:
do we know for how long we are caching?

Reply:
same as current setting 1440 seconds - 4 hours. A lot tbh. Also, check my comment on DefaultUserRepository

Reply:
mmm yes a bit too much